### ISSUE TITLE
Offline/Error handling of signout with signoutFromBrowser 

### ISSUE THREAD

**Issue by t-hand** [AzureAD/microsoft-authentication-library-for-objc#1133]
Hello,

I've encountered the following problem/potential bug.

When `signout` with `signoutFromBrowser` is called in a case where the application has no internet connection the user has to dismiss the presented AuthenticationSession view. In this case the user is logged out locally but all potential SSO cookies still exist and calling `acquireToken` without `promptType = .login` will automatically log in the user again when connectivity is restored.

Despite this logout only being partial the `signout` method returns success with no error.
Without (ab)using `MSALWebAuthDidFailNotification` it seems impossible to determine if the user was really logged out and inform him of a potential security risk.
The problem would presumably also apply if there is any other reason why the signout endpoint could not be reached. 

The `ASWebAuthenticationSessionErrorCodeCanceledLogin` is recognised correctly in `MSIDASWebAuthenticationSessionHandler` and passed through to `MSIDOIDCSignoutRequest`.
There it is detected by
```
if (error && !([error.domain isEqualToString:MSIDErrorDomain] && error.code == MSIDErrorUserCancel))
```
and dismissed by reporting success without error to the completion handler.

Why is an obvious Error while logging out handled that way?
What would be the correct way to ensure signoutFromBrowser was successful?

Tested with `MSALWebviewType = .default` on iOS14.
Thanks!

**Comment by ameyapat** [comment]
@t-hand In this case, we make a call to logout endpoint and the server is responsible to clear any cookies via the response. We currently don't have support to detect if the signout operations performed by the server were successful or not. From client side we will perform operations like clearing keychain entries and caches from the device.

We don't handle cookies in MSAL and leave the browser to do it. You may use an embedded webview  to handle this edge case but that might also involve having your own SSO and cookie handling mechanism.

**Comment by t-hand** [comment]
I fully understand that the browser is responsible for deleting the Cookies and that MSAL can only open the logout url so please let me rephrase my original questions.

What is the correct way to detect that the user closed the `ASWebAuthenticationSession`? `ASWebAuthenticationSessionErrorCodeCanceledLogin` is not handed over to the `signout` completion block which ends in a state that our client deems a security risk. Is this a bug or by design?

One possible way i found was by listening for `MSALWebAuthDidFailNotification` but this is only documented for the use of a custom webview.
Is it safe to rely on this notifications when using `ASWebAuthenticationSession` or is this behaviour subject to change.

Using a custom webview is no option for us.

**Comment by aherciya** [comment]
Hi @t-hand. Unfortunately this is by design and the issue is we don't directly control cookies in the browser.

**Comment by t-hand** [comment]
@aherciya: Thanks for your response. As pointed out before, we are aware that MSAL has no control over the content of the browser and cookies.
Assuming that you mean that the success state in case of an active user cancelation (user taps "Cancel" button in ASWebAuthenticationSession) is by design, could you please confirm that sending the `MSALWebAuthDidFailNotification` notification is also by design for `ASWebAuthenticationSession`s and not subject to change so we can handle this state on our own?

The documentation (https://docs.microsoft.com/en-us/azure/active-directory/develop/customize-webviews) only speaks of these notifications in context of a custom WebView.

**Comment by aherciya** [comment]
@t-hand As of now there aren't any plans to make changes, but we follow semver, so you should be informed when there will be a major update. Using ASWebAuthenticationSessions with MSALWebAuthDidFailNotification is not by design.
